### DISCUSSION TITLE
ELN display

### DISCUSSION THREAD

**Discussion by Docbroke** [leo-arch/clifm#65] [General]
Instead of current
```
8
9
10
11
12
```
we may use
```
08
09
10
11
12
``` 
or
```
 8
 9
10
11
```
to make it look uniform.

btw., clifm is now in official alpine linux repositories.

**Comment by leo-arch** [comment]
Hi @Docbroke. I remember I tried something like this once, but I didn't like the result when you list lots of files: 01 is readable, but 0001 could be a bit annoying. Spaces could be used as well to pad ELN's. So, instead of "0001" we could have "   1", or maybe, right padding it: "1   ".

I guess making this optional might work. Something like `ElnPad` (in the config file) taking the following values: `none`, `zero`, `leftspace`, and `rightspace`.

**Comment by Docbroke** [comment]
Padding will do. Purpose is to have file/directory names displayed symmetrically.

**Reply by leo-arch** [reply]
Great. Working on it.

**Comment by leo-arch** [comment]
Three ELN padding modes (no padding should be an option too):

![zeropad](https://user-images.githubusercontent.com/21303106/138731045-382ae1cb-8769-42ac-b2ac-d8e5a9ef8d8d.png)
![leftspacepad](https://user-images.githubusercontent.com/21303106/138731073-44a3a7ff-eeb5-49f8-bc53-7b978c614dc5.png)
![rightspacepad](https://user-images.githubusercontent.com/21303106/138731099-965bbc25-e1ee-4629-b3f6-6558da98f956.png)

**Reply by Docbroke** [reply]
Great. It is good to have options, however I think padding on left looks good as well as mathematically correct. Zeros may actually confuse and padding on right is mathematically wrong ( 1 in tenth's place).

**Reply by leo-arch** [reply]
So, do you think we should make leftspace padding (second picture) the default value?
Be it as it may, we still need to consider how padding looks when you have hundreds or thousands of files.

EDIT: I guess autocommands could be used for this purpose: for example, if more than 999 files are listed, I don't want any padding for that folder. Just thinking out loud.

**Reply by Docbroke** [reply]
Yes, I feel left space padding is good enough default.

**Reply by leo-arch** [reply]
Cool. This new feature should be in master in a few hours.

**Reply by leo-arch** [reply]
Well, this thing should be working now. Give it a try and let me know. There's a new option in the config file: `ELNPad` (values: 0=none, 1=zeroes, 2=spaces on the left, and 3=spaces on the right).

**Reply by Docbroke** [reply]
Thanks.
This is perfect. With a quick try, ELNPad=2 looks good to me. 1(zeroes) is also fine for small number of files. 3(right) looks odd to me.

**Reply by leo-arch** [reply]
Great. As you said, it's always good to have options.

**Comment by leo-arch** [comment]
> clifm is now in official alpine linux repositories

:rocket:

**Reply by Docbroke** [reply]
I asked for that https://gitlab.alpinelinux.org/alpine/aports/-/issues/13111
And they responded within 2 days.

That is reason I am here. I have recently started using alpine as my main os.

**Reply by leo-arch** [reply]
Ohhh, it was you! Thank you very much! Having CliFM in official repos is an important way of making this little monster grow.
